test/unit/detail/version_check_test.hpp

```diff
@@ -208,85 +195,65 @@ TEST_F(sanity_checks, cookie)
-                  << "Updates done here: https://github.com/OpenMS/usage_plots/blob/master/seqan_versions.txt"
-                  << std::endl;
+                  << "Updates done here: https://github.com/OpenMS/usage_plots/blob/master/seqan_versions.txt\n";
     }
 }
 
 //------------------------------------------------------------------------------
 // version checks
 //------------------------------------------------------------------------------
 
-TEST_F(version_check, option_on)
+TEST_F(version_check_test, option_on)
 {
-    char const * argv[3] = {app_name.c_str(), OPTION_VERSION_CHECK, OPTION_ON};
-
-    auto [out, err, app_call_succeeded] = simulate_parser(3, argv);
+    auto [out, err, app_call_succeeded] = simulate_parser(OPTION_VERSION_CHECK, OPTION_ON, "-f");
 
     EXPECT_EQ(out, "");
     EXPECT_EQ(err, "");
 
     // no timestamp is written since the decision was made explicitly
     if (app_call_succeeded)
-    {
         EXPECT_TRUE(std::filesystem::exists(app_version_filename()));
-    }
     else
-    {
-        std::cout << "App call did not succeed (server offline?) and could thus not be tested.\n";
-    }
+        GTEST_SKIP() << "App call did not succeed (server offline?) and could thus not be tested.";
 
     EXPECT_TRUE(remove_files_from_path()); // clear files again
 }
 
 // Note that we cannot test interactiveness because google test captures std::cin and thus
 // sharg::detail::input_is_terminal() is always false
-TEST_F(version_check, option_implicitely_on)
+TEST_F(version_check_test, option_implicitely_on)
 {
-    char const * argv[2] = {app_name.c_str(), "-f"};
-
-    auto [out, err, app_call_succeeded] = simulate_parser(2, argv);
+    auto [out, err, app_call_succeeded] = simulate_parser("-f");
 
     EXPECT_EQ(out, "");
-    EXPECT_EQ(err,
-              "\n#######################################################################\n"
-              "   Automatic Update Notifications\n"
-              "#######################################################################\n"
-              " This app performs automatic checks for updates. For more information\n"
-              " see: https://docs.seqan.de/sharg/main_user/about_update_notifications.html\n"
-              "#######################################################################\n\n");
+    EXPECT_EQ(err, "");
 
     // make sure that all files now exist
     EXPECT_TRUE(std::filesystem::exists(app_timestamp_filename())) << app_timestamp_filename();
-    EXPECT_TRUE(std::regex_match(read_file(app_timestamp_filename()), timestamp_regex));
+    EXPECT_TRUE(std::regex_match(read_first_line(app_timestamp_filename()), timestamp_regex));
 
-    if (app_call_succeeded)
-    {
-        EXPECT_TRUE(std::filesystem::exists(app_version_filename()));
-    }
-    else
-    {
-        std::cout << "App call did not succeed (server offline?) and could thus not be tested.\n";
-    }
+    EXPECT_FALSE(app_call_succeeded);
```

Comment:
The nature of the test changed? If the check is implicitely on, why do we not expect the notification anymore? (and what does this have to do with the actual change to use the test fixture)

Reply:
So, this test was never executed.
At some point we cleaned up the `sharg::detail::input_is_terminal()` etc.

When we are not running in an interactive terminal, we don't ask, we assume no.